### PR TITLE
Update submodules, update Squirrel registration

### PR DESCRIPTION
**Pull request by cticenhour**
Given https://github.com/arfc/squirrel/pull/5 and the recent MOOSE libMesh update, this PR updates both MOOSE and Squirrel submodules to their most current master branches, and fixes up `ZapdosApp.C` to use the new `registerAll` function in `SquirrelApp`. 

@lindsayad I promise this is my last maintenance PR for a while :wink:

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -3,4 +3,4 @@
 	url = https://github.com/arfc/squirrel
 [submodule "moose"]
 	path = moose
-	url = ../../idaholab/moose
+	url = https://github.com/idaholab/moose
```
Comment (lindsayad) [inline .gitmodules]: Why this change? This is here so that people who clone zapdos with ssh would do the same for submodules

Comment (cticenhour) [reply .gitmodules]: I have always been confused about this convention, as I thought it was just for convenience. Because of that, I always preferred typing the URL so I could click through to the appropriate code for the submodule if I needed to look something up online. 

If this is for SSH, why does only MOOSE use a different convention and not Squirrel?

Comment (lindsayad) [reply .gitmodules]: Because I didn't know about the relative path usage when I added the squirrel submodule and for some reason I was too lazy to change one other line when I added the moose submodule. I'd like to have this changed back, and maybe if you're less lazy than I am you can change the squirrel url 😃

Comment (cticenhour) [reply .gitmodules]: Will do!
